x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -99,6 +99,17 @@ export class Session {
     this.crypto = nodeCrypto({ encryptionKey: this.options.config.encryptionKey });
   }
 
+  /**
+   * Extracts session id for the specified request.
+   * @param request Request instance to get session value for.
+   */
+  async getSID(request: KibanaRequest) {
+    const sessionCookieValue = await this.options.sessionCookie.get(request);
+    if (sessionCookieValue) {
+      return sessionCookieValue.sid;
+    }
```

Comment:
I find `null` a cause of unintended bugs and behaviours and the distinctions between "not set" (`undefined`) and "set but empty" (`null`) to be rarely important. 

e.g. `null` can cause bugs when using optional parameters and default values since the default value will not be used. `null` is also of type object which is weird and can cause bugs so I rarely use it.

Having said that, I think in this case, we really are dealing with the "not set" scenario so I think `undefined` is correct. 

Why are you using `null` for the other methods?

Reply:
> Why are you using null for the other methods?

**TL;DR:** We don't have any guidelines on that yet and even though I have a slightly stronger opinion on `getCurrentUser` and `Session.get` as explained below, I don't have the same strong reasons to use `null` in this particular case except for consistency. Both would work for me and I trust your judgement here :+1: 

I _think_ that we historically treated `null` not as `set but empty`, but more like [an intentional absence of any object value](https://www.ecma-international.org/ecma-262/#sec-null-value), like the user or session objects are absent in a particular context, it's expected and intentional, and we explicitly manifest that with `null`. As a side benefit, when you `return null;` you explicitly define an exit point, where `undefined` can be either intentional or not (e.g. forgotten `return` statement). 

Regarding default and optional parameters, I'd say it's more about personal preference, I find the code that is explicit about using default and optional parameters a little bit easier to read and understand, and type system will prevent unintentional behavior in case parameter cannot be `null`.

Reply:
I think the definition makes sense* then. For object values use `null`, for primitives use `undefined`. That also explains why null is of type object. 

* I'm using "sense" in the loosest meaning of the word here since most other programming languages have a single type the denote absence of a value and get by just fine 🤷‍♀️